client/my-app/src/components/Workflow.tsx

```diff
@@ -0,0 +1,142 @@
+"use client";
+import { useEffect, useState } from "react";
+import { Input } from "@/components/ui/input";
+import { Check } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import { Card } from "@/components/ui/card";
+import { Badge } from "@/components/ui/badge";
+import { useSearchParams } from "next/navigation";
+import { useRouter } from "next/navigation";
+
+
+export default function Workflow({}) {
+  const [showEmailInput, setShowEmailInput] = useState(false);
+  const [email, setEmail] = useState("");
+  const [isDeploying, setIsDeploying] = useState(false);
+  const [repoName, setRepoName] = useState<string | null>(null);
+  const [owner, setOwner] = useState<string | null>(null);
+  const [accessToken, setAccessToken] = useState<string | null>(null);
+  const [repoUrl, setRepoUrl] = useState<string | null>(null);
+
+  const baseUrl = process.env.NEXT_PUBLIC_BASE_URL;
+  const user_id = localStorage.getItem("user_id") || "";
+
+  const searchParams = useSearchParams();
+  const router = useRouter();
+
+  const extractOwnerAndRepo = (repoUrl: string) => {
+    const url = new URL(repoUrl);
+    const pathParts = url.pathname.split("/");
+
+    const owner = pathParts[1];
+    const repoName = pathParts[2];
+
+    return { owner, repoName };
+  };
+
+  useEffect(() => {
+    const repoUrlFromQuery = searchParams.get("repo");
+    if (repoUrlFromQuery) {
+      setRepoUrl(repoUrlFromQuery);
+      const { owner, repoName } = extractOwnerAndRepo(repoUrlFromQuery);
+      setOwner(owner);
+      setRepoName(repoName);
+    }
+    const token = localStorage.getItem("access_token");
+    if (token) {
+      setAccessToken(token);
+    }
+  }, [searchParams]);
+
```

Comment:
why searchParams is in the dependency array ..

rather " const repoUrlFromQuery = searchParams.get("repo"); " should come outside the useEffect and " repoUrlFromQuery " should come in the dependency array...

here is the reason bcz ... everytime when the repoUrlFromQuery changes then we need to run the useEffect